addons/account/models/account_account.py

```diff
@@ -255,14 +281,16 @@ def _check_journal_consistency(self):
                 account=account.display_name
             ))
 
-    @api.constrains('company_id')
+    @api.constrains('company_ids')
     def _check_company_consistency(self):
-        for company, accounts in tools.groupby(self, lambda account: account.company_id):
-            if self.env['account.move.line'].search_count([
+        if self.filtered(lambda a: not a.company_ids):
+            raise UserError(_("Every account must belong to at least one company!"))
```

Comment:
`_constrain_at_least_one_company`?

Reply:
true :+1: 
I prefer then to delete `constrain_at_least_one_company` and move that code here.